### PR TITLE
npctrade: trade multiple items at once

### PR DESCRIPTION
**Pull request by mlangsdorf**
#### Summary
```SUMMARY: Features "npctrade: trade multiple items at once"```

#### Purpose of change
There's no open issue for this feature, even though it's one of the oldest TODOs in the code.

Trade stacks of item at once.

#### Describe the solution
Refactor npctrade to have a slightly more sensible structure, including a class to manage the window.

Rewrite the code to exchange inventories to be simpler and hopefully faster.

Rework npctrade to pass slice sizes.  For non-count by charge items with a slice size of 2 or more, display the stack size in the trade window.  When the player selects an item with a charge count more than 1 or stack size more than 1, add a prompt for how many items they want to trade.  Deselecting the item deselects all items in that stack.
    
Transfer items between the player and NPC as specified.

#### Describe alternatives you've considered
I'm using use_charges() to delete items with charges from inventory, which seems valid, and use_amount( 1 ) in a loop to delete multiple copies of the same item from inventory, which seems hacky.  A better solution would be appreciated; this works.

My test bionic gangster has 2 different "Glock 17 Magazine (15/15)" in his inventory - one in hand, one in gun.  The first attempt to sell the one from inventory doesn't remove it from inventory, I don't know why, but subsequent attempts work.

#### Additional context
Built on top of #31245 
multi-trade
"I'll take the dive bag and 50 Merch for 6 meat jerky, the pack of firecrackers, and a plutonium fuel cell."

multi-trade2
"Deal!"

### CODE REVIEW

**Comment by davidpwbrown** [comment]
This is excellent, I will try and review what I can tomorrow.

File: data/json/npcs/tacoma_ranch/NPC_ranch_bartender.json [hunk 1]
```diff
@@ -48,25 +48,21 @@
...
         "text": "[$8] I'll take a beer",
         "topic": "TALK_DONE",
-        "condition": { "u_has_cash": 800 },
         "effect": { "u_buy_item": "beer", "container": "bottle_glass", "count": 2, "cost": 800 }
       },
       {
         "text": "[$10] I'll take a shot of brandy",
         "topic": "TALK_DONE",
-        "condition": { "u_has_cash": 1000 },
         "effect": { "u_buy_item": "beer", "container": "bottle_glass", "cost": 1000 }
```
Comment (davidpwbrown) [inline data/json/npcs/tacoma_ranch/NPC_ranch_bartender.json]: "u_buy_item": "beer" , but buying spirits?

File: data/json/npcs/tacoma_ranch/NPC_ranch_bartender.json [hunk 2]
```diff
@@ -48,25 +48,21 @@
...
       },
       {
         "text": "[$10] I'll take a shot of rum",
         "topic": "TALK_DONE",
-        "condition": { "u_has_cash": 1000 },
         "effect": { "u_buy_item": "beer", "container": "bottle_glass", "cost": 1000 }
```
Comment (davidpwbrown) [inline data/json/npcs/tacoma_ranch/NPC_ranch_bartender.json]: "u_buy_item": "beer" , but buying spirits?

File: data/json/npcs/tacoma_ranch/NPC_ranch_bartender.json [hunk 3]
```diff
@@ -48,25 +48,21 @@
...
       },
       {
         "text": "[$12] I'll take a shot of whiskey",
         "topic": "TALK_DONE",
-        "condition": { "u_has_cash": 1200 },
         "effect": { "u_buy_item": "beer", "container": "bottle_glass", "cost": 1200 }
```
Comment (davidpwbrown) [inline data/json/npcs/tacoma_ranch/NPC_ranch_bartender.json]: "u_buy_item": "beer" , but buying spirits?

File: doc/NPCs.md [hunk 4]
```diff
@@ -441,11 +441,11 @@ Effect | Description
...
+`u_buy_item: item_string`, (*optional* `cost: cost_num`, *optional* `count: count_num`, *optional* `container: container_string`) | The NPC will give your character the item or `count_num` copies of the item, contained in container, and will subtract `cost_num` from `op_of_u.owed` if specified.  If the `op_o_u.owed` is less than `cost_num`, the trade window will open and the player will have to trade to make up the difference; the NPC will not give the player the item unless `cost_num` is satisfied.<br/>If cost isn't present, the NPC gives your character the item at no charge.
+`u_sell_item: item_string`, (*optional* `cost: cost_num`, *optional* `count: count_num`) | Your character will give the NPC the item or `count_num` copies of the item, and will add `cost_num` to the NPC's `op_of_u.owed` if specified.<br/>If cost isn't present, the your character gives the NPC the item at no charge.<br/>This effect will fail if you do not have at least `count_num` copies of the item, so it should be checked with `u_has_items`.
```
Comment (davidpwbrown) [inline doc/NPCs.md]:
```suggestion
`u_sell_item: item_string`, (*optional* `cost: cost_num`, *optional* `count: count_num`) | Your character will give the NPC the item or `count_num` copies of the item, and will add `cost_num` to the NPC's `op_of_u.owed` if specified.<br/>If cost isn't present, then your character gives the NPC the item at no charge.<br/>This effect will fail if you do not have at least `count_num` copies of the item, so it should be checked with `u_has_items`.
```

**Comment by mlangsdorf** [comment]
jenkins rebuild

File: src/npctrade.cpp [hunk 5]
```diff
@@ -67,244 +108,322 @@ std::vector<item_pricing> init_selling( npc &p )
...
-                mvwprintz( w_head, 3, ( TERMX - deal.length() ) / 2, cost < 0 ? c_light_red : c_light_green,
-                           deal.c_str() );
+                int keychar = i - offset + 'a';
+                if( keychar > 'z' ) {
+                    keychar = keychar - 'z' - 1 + 'A';
+                }
+                trim_and_print( w_whose, i - offset + 1, 1, win_w, color, "%c %c %s",
+                                static_cast<char>( keychar ), ip.selected ? '+' : '-', itname );
+#if defined(__ANDROID__)
+                ctxt.register_manual_key( keychar, itname );
```
Comment (ZhilkinSerg) [inline src/npctrade.cpp]: No input context were defined previously, so we should probably add before `for` cycle:

```
#if defined(__ANDROID__)
        input_context ctxt( "NPC_TRADE" );
#endif
```

**Pull request merged**